app/controllers/orders_controller.rb

```diff
@@ -24,6 +24,15 @@ def show
   end
 
   def update
+    if @current_user.try(:instance_of?, Driver)
+      return render json: { 'error' => 'You are not allowed  to change an order' }, status: 401
```

Comment:
@domenic99, yes, he isn't. But it's because he is driver, not because he is not logged in :wink: